src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -142,13 +143,52 @@ object ScalafixPlugin extends AutoPlugin {
       Invisible
     )
 
+  private val adaptSbtResolvers = Def.task {
+    val credentialsByHost = Credentials
+      .allDirect(credentials.value)
+      .map(dc =>
+        dc.host -> coursierapi.Credentials.of(
+          dc.userName,
+          dc.passwd
+        )
+      )
+      .toMap
+
+    resolvers.value.flatMap(resolver => {
+      val adaptedRepo = CoursierRepoResolvers.repository(
+        resolver,
+        stdoutLogger,
+        credentialsByHost
+      )
+      if (adaptedRepo.isEmpty) {
+        stdoutLogger.warn(
+          s"Defined resolver $resolver cannot be converted to coursier repository, it will be ignored by scalafix."
+        )
+      }
+      adaptedRepo
+    })
+  }
+
   private lazy val cachingStyle = {
     val useLastModifiedCachingStyle =
       sys.props.get("sbt-scalafix.uselastmodified") == Some("true")
     if (useLastModifiedCachingStyle) FileInfo.lastModified
     else FileInfo.hash
   }
 
+  private lazy val defaultScalafixResolvers =
+    // Repository.defaults() defaults to Repository.ivy2Local() and Repository.central(). These can be overridden per
+    // env variable, e.g., export COURSIER_REPOSITORIES="ivy2Local|central|sonatype:releases|jitpack|https://corporate.com/repo".
+    // See https://github.com/coursier/coursier/blob/master/modules/coursier/jvm/src/main/scala/coursier/PlatformResolve.scala#L19-L68
+    // and https://get-coursier.io/docs/other-repositories for more details.
+    // Also see src/sbt-test/sbt-scalafix/scalafixResolvers/test for a scripted test preserving this behavior.
```

Comment:
If we still have to resolve `Credentials`, I think since it will inevitably make adapt sbt resolvers a task, I think remove this wouldn't help the implementation.